algebird-test/src/main/scala/com/twitter/algebird/scalacheck/Gen.scala

```diff
@@ -69,4 +69,20 @@ object gen extends ExpHistGen with IntervalGen {
   def genSSManySpaceSaver: Gen[SpaceSaver[String]] =
     Gen.nonEmptyListOf(genFixedSSOneSpaceSaver).flatMap(l => l.reduce(_ ++ _))
 
+  def genCorrelation: Gen[Correlation] =
```

Comment:
this never generates 0, right? (a count of zero)? A way that I like that might actually be simpler is to generate according to how this is used:
```scala
lazy val genCorrelation: Gen[Correlation] = {
  val recur = Gen.lzy(genCorrelation)
  
  // we can start with any pair of numbers:
  val genClose: Gen[Correlation] = for {
    x <- choose(-1000, 1000)
    delta <- choose(-100.0, 100.0)
   } yield Correlation((x, x + delta))

  val genUncorr: Gen[Correlation] = for {
    x <- choose(-1e20, 1e20)
    y <- choose(-1e20, 1e20)
   } yield Correlation((x, y))

  val genSum = Gen.zip(recur, recur).map { case (a, b) => a + b }
  val genDiff = Gen.zip(recur, recur).map { case (a, b) => a - b }
  val genNeg = recur.map(_.negate)
  
  // now return with a low probability of choosing the branching cases:
  Gen.frequency(
    (5, genClose),
    (5, genUncorr),
    (2, genNeg),
    (1, Correlation.zero),
    (1, genSum),
    (1, genDiff))
}
```
In this style, you are generating values that are built up from formulas of using the data.

Reply:
That's true, although it _does_ get verified when validating `Group` laws. Although my guess is that you prefer this for tests written by consumers of this library, right?

Reply:
I ended up needing to make exponent of the range of Doubles be `10` instead of `20`. Otherwise, it was too easy for variance to become `Infinity` which often ends up making `c2` be `NaN`, which in turn makes the resulting `Correlation` instance unequal to itself.